PyPoE/poe/file/bundle.py

```diff
@@ -449,23 +466,25 @@ def get_hash(self, path: Union[str, bytes], type: PATH_TYPES = None) -> int:
 
         Returns
         -------
-        Calculated 64bit FNV1a hash value
+        Calculated 64bit hash value
         """
-        if isinstance(path, str):
-            path = path.encode('utf-8')
-        elif not isinstance(path, bytes):
-            raise TypeError('path must be a string')
-
-        if path.endswith(b'/'):
-            if type is None:
-                type = PATH_TYPES.DIR
-            path = path.strip(b'/')
-        # If type wasn't set before, assume this is a file
-        if type == PATH_TYPES.FILE or type is None:
-            path = path.lower()
-        path += b'++'
-
-        return fnv1a_64(path)
+        if self.hash_algorithm == HASH_ALGORITHM.FNV1A64:
+            if isinstance(path, str):
+                path = path.encode('utf-8')
+            elif not isinstance(path, bytes):
+                raise TypeError('path must be a string')
+            if path.endswith(b'/'):
+                if type is None:
+                    type = PATH_TYPES.DIR
+                path = path.strip(b'/')
+            # If type wasn't set before, assume this is a file
+            if type == PATH_TYPES.FILE or type is None:
+                path = path.lower()
+            path += b'++'
```

Comment:
Samey behaviour compared to the other file where `_hash_path_3_21_2` is implemented. This could perhaps already warrant a small sanitizer helper seeing as the order of actions is `type check > strip > case change`.